channel.go

```diff
@@ -1,14 +1,33 @@
 package thingfulx
 
-import (
-	"time"
-)
+import "time"
 
 // Channel represents an individual data channel being published by a Thing out
 // there in the world somewhere.
 type Channel struct {
-	ID         string    `json:"id"`
-	Value      string    `json:"value"`
+	ID           string            `json:"id"`
+	Metadata     []ChannelMetadata `json:"metadata"`
+	Observations []Observation     `json:"observations"`
+	Data         Data              `json:"data"`
+}
+
+// ChannelMetadata is an object which contains any metadata required to
+// semantically descibe a measurement collected by a Thing.
+type ChannelMetadata struct {
+	Prop string `json:"property"`
+	Val  string `json:"value"`
+}
```

Comment:
It would be shipped to the semantic search module, and you're right given the lack of a rigorous json-ld style @context to define the expansions, we would need to provide a poor mans implementation of that which probably would like some sort of simple in memory mapping/lookup object.